src/Tribe/Repository/Query_Filters.php

```diff
@@ -616,16 +616,6 @@ protected function where_field_is( $where, WP_Query $query, $field, $prepare = '
 		return $where;
 	}
 
-	/**
-	 * Clean up before the object destruction.
-	 *
-	 * @since 4.7.19
-	 */
-	public function __destruct() {
```

Comment:
The issue is the existence of the `__destruct()` to begin with.

To clarify with an anecdote:

- If I put a return at the beginning of the destruct it runs just as slow. The array / remove filters isn't the issue.
- If I remove the destruct altogether, the script is about 5s~ faster (on our events admin page for a customers database). 

I did some googling and I saw people pointing out some nuance to the GC feature and circular reference memory cleanup. My assumption is the circular reference cleanup is sidestepped without the destruct function, because it just clears up the whole scripts memory instead of doing individual object reference checks.